lib/plugins/package/lib/packageService.js

```diff
@@ -33,11 +33,10 @@ module.exports = {
       .getServerlessConfigFilePath(this.serverless)
       .then(configFilePath => {
         const packageExcludes = this.serverless.service.package.exclude || [];
-        // add local service plugins Path
-        const pluginsLocalPath = this.serverless.pluginManager.parsePluginsObject(
-          this.serverless.service.plugins
-        ).localPath;
-        const localPathExcludes = pluginsLocalPath ? [pluginsLocalPath] : [];
+        // add local service plugins Paths
+        const pluginsLocalPaths = this.serverless.pluginManager
+          .getLocalPluginsPaths(this.serverless.service.plugins)
+          .map(pluginsLocalPath => `${pluginsLocalPath}/**`);
```

Comment:
This collection may contain both _directory_ and _file_ paths. e.g. when local plugin referenced as  `./plugin`, may actually point `./plugin.js`. Therefore treating all paths as directories won't work well.

It's hard to have 100% bulletproof solution for that, but I think what could work best is:

- Agree that `getLocalPluginsPaths` may return paths to directories, and files. To make distinction easy, let's ensure that all directory paths are postfixed with `/` (or system path separator - we need to be careful with Windows here) 
- In `getLocalPluginsPaths` return paths as follows
  - If there's a custom local path, return it with postfixed `/` (as it's clearly a directory path)
  - Resolve final paths to plugins referenced with relative paths (starting with `./`), as it's done here: https://github.com/serverless/serverless/blob/5a444c415ce31b2c219be47390be165a8da233ea/lib/classes/PluginManager.js#L26. If path as input into configuration, points a directory (in which main plugin module lives) then return it postfixed with `/`, if path points a plugin file, then return it with fully resolved file extension.

Having above ensure glob rules are applied differently to directories and files

Reply:
> However, if this file locally requires utils.js, it will not be identified as a file to be excluded and will be shipped with lambda package. Correct ?

Yes, agreed, it's exactly the reason I mentioned _it's hard to have 100% bulletproof solution_ :)

> for each item of this list, one exact file resolution blob ${item}.js and one directory resultion blob ${item}/**

This may work for most of the cases, still is immune to one flaw. Imagine user having both `foo.js` and `foo` folder which is expected to land in lambda. With above we will not package `foo` folder, and that will be clearly a bug on our side.

Of course it's an edge case, but one we can avoid by resolving the main plugin module path, and reasoning against that. It'll be nice to not open door for false positives.